src/pages/requests/index.tsx

```diff
@@ -50,55 +52,83 @@ export function Requests() {
   // TODO: consider moving this to a component higher up in the render tree
   // TODO: consider simplifying filterContext API
   useEffect(() => {
-    if (suppliesDict) {
-      addFilter({
-        filterName: "Categories",
-        filterItems: Object.values(suppliesDict).map((category): FilterItem => ({ id: category.id, selected: false, text: category.name })),
-        active: false,
-        singleValueFilter: false,
-      });
-    }
+    if (isAidRequestsLoaded && isSuppliesLoaded && isLocationsLoaded) {
+      if (suppliesDict) {
+        addFilter({
+          filterName: "Categories",
+          filterItems: Object.values(suppliesDict).map(
+            (category): FilterItem => ({ id: category.id, selected: false, text: category.name })
+          ),
+          active: false,
+          singleValueFilter: false,
+        });
+      }
 
-    if (locationDict) {
-      addFilter({
-        filterName: "Cities",
-        filterItems: Object.values(locationDict).map((location): FilterItem => ({ id: location.id, selected: false, text: location.name })),
-        active: false,
-        singleValueFilter: false,
-        hasSearch: true,
-      });
-    }
+      if (locationDict) {
+        addFilter({
+          filterName: "Cities",
+          filterItems: Object.values(locationDict).map(
+            (location): FilterItem => ({ id: location.id, selected: false, text: location.name })
+          ),
+          active: false,
+          singleValueFilter: false,
+          hasSearch: true,
+        });
+      }
+
+      if (!isEmpty(aidRequestsGroupedByDate)) {
+        const uniqueDatesSorted = uniq(keys(aidRequestsGroupedByDate)).sort(sortDates);
+
+        addFilter({
+          filterName: "Dates",
+          filterItems: uniqueDatesSorted.map(
+            (date, i): FilterItem => ({ id: date, selected: i === uniqueDatesSorted.length - 1, text: date })
+          ),
+          active: false,
+          singleValueFilter: true,
+        });
+      }
 
-    if (!isEmpty(aidRequestsGroupedByDate)) {
-      const uniqueDatesSorted = uniq(keys(aidRequestsGroupedByDate)).sort(sortDates);
-
-      addFilter({
-        filterName: "Dates",
-        filterItems: uniqueDatesSorted.map(
-          (date, i): FilterItem => ({ id: date, selected: i === uniqueDatesSorted.length - 1, text: date })
-        ),
-        active: false,
-        singleValueFilter: true,
-      });
+      // Now that the filters are initialized, it's safe to apply any filters from query params
```

Comment:
**thought**
It's actually not obvious from the code, I guess. 
Checking explicitly if filters are initialized in a separate `useEffect` and `setCanApplyQuery(true)` there would make it safe and obvious. 

It would also decouple filters initialisation from query logic.